uint32 encoder_position
time time_received
